src/test/kotlin/no/nav/syfo/personstatus/infrastructure/kafka/ArbeidsuforhetVurderingConsumerSpek.kt

```diff
@@ -0,0 +1,94 @@
+import no.nav.syfo.personstatus.PersonoversiktStatusService
+import no.nav.syfo.personstatus.db.createPersonOversiktStatus
+import no.nav.syfo.personstatus.db.getPersonOversiktStatusList
+import no.nav.syfo.personstatus.domain.PersonOversiktStatus
+import no.nav.syfo.personstatus.infrastructure.database.PersonOversiktStatusRepository
+import no.nav.syfo.testutil.ExternalMockEnvironment
+import no.nav.syfo.testutil.UserConstants
+import no.nav.syfo.testutil.dropData
+import org.amshove.kluent.shouldBeEqualTo
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+
+class ArbeidsuforhetVurderingConsumerSpek : Spek({
+    val externalMockEnvironment = ExternalMockEnvironment.instance
+    val database = externalMockEnvironment.database
+    val kafkaConsumer = mockk<KafkaConsumer<String, ArbeidsuforhetVurderingRecord>>()
+    val personOppgaveRepository = PersonOversiktStatusRepository(database = database)
+    val personoversiktStatusService = PersonoversiktStatusService(
+        database = database,
+        pdlClient = externalMockEnvironment.pdlClient,
+        personoversiktStatusRepository = personOppgaveRepository
+    )
+
+    val arbeidsuforhetVurderingConsumer = ArbeidsuforhetVurderingConsumer(
+        personoversiktStatusService = personoversiktStatusService,
+    )
+    val arbeidsuforhetVurderingRecord = generateArbeidsvurderingRecord(
+        personIdent = PersonIdent(UserConstants.ARBEIDSTAKER_FNR),
+        createdAt = OffsetDateTime.now(),
+    )
+    beforeEachGroup { database.dropData() }
+    beforeEachTest {
+        every { kafkaConsumer.commitSync() } returns Unit
+    }
+
+    afterEachTest {
+        database.dropData()
+        clearMocks(kafkaConsumer)
+    }
+
+    describe("pollAndProcessRecords") {
+        it("consumes arbeidsuforhet vurdering and updates personoversikt status") {
+            val newPersonOversiktStatus = PersonOversiktStatus(fnr = UserConstants.ARBEIDSTAKER_FNR)
+            database.connection.use { connection ->
+                connection.createPersonOversiktStatus(
+                    commit = true,
+                    personOversiktStatus = newPersonOversiktStatus,
+                )
+            }
+
+            kafkaConsumer.mockPollConsumerRecords(
+                recordValue = arbeidsuforhetVurderingRecord,
+                topic = "teamsykefravr.arbeidsuforhet-vurdering",
+            )
+
+            arbeidsuforhetVurderingConsumer.pollAndProcessRecords(kafkaConsumer = kafkaConsumer)
+
+            verify(exactly = 1) {
+                kafkaConsumer.commitSync()
+            }
+            val personStatus = database.getPersonOversiktStatusList(fnr = UserConstants.ARBEIDSTAKER_FNR).single()
+            personStatus.fnr shouldBeEqualTo arbeidsuforhetVurderingRecord.personident
```

Comment:
Burde vi ikke også teste at man oppdaterte arbeidsuførhet-feltet? Fordi nå tester man egentlig bare at det på linje 55-60 fungerer, ettersom begge har samme FNR?
`personStatus.isAktivArbeidsuforhetVurdering shouldBeEqualTo true`
`isAktivArbeidsuforhetVurdering` ligger kanskje ikke på domeneobjektet, men burde det ikke det?

Reply:
Jo, selvfølgelig. Her var jeg litt kjapp. Bra sett 🙌

Reply:
Hvilket domeneobjekt tenker du på ? :)

Reply:
Det som heter `PersonOversiktStatus` eller noe sånt. For nå lever bare feltet i `PPersonOversiktStatus`?

Reply:
Aha, ja den er også lagt til i `PersonOversiktStatus` 😁